CompilerSource/compiler/event_reader/event_parser.h

```diff
@@ -53,6 +53,11 @@ enum p_type {
 #include <vector>
 #include <map>
 
+using std::map;
```

Comment:
"using" in headers will leak into header including this header. I know this isn't as a big issue in the compiler but it is still a bad habbit.

Reply:
Cleaning up this header is not something I intend to do in this pull request. Attempting to remove this would create dozens of useless line changes.

Reply:
wasnt't asking that you clean it just that you don't dirty it further :)

Reply:
As it stands (at HEAD), the file depends on its caller putting this data in the global namespace for it. As you have pointed out, this is a code smell. I have modified the file so that now it ensures its global namespace pollution itself. This improves the header by making it possible to include it without being charged with the task of polluting the global namespace on your own. 

Regardless, this is not going to be a bureaucratic process in which everyone modifying a file must feel obligated to "fix" the file to fit a more hip and modern understanding of how it should look. Or, as you'd argue, to "not dirty it further." I always point out to contributors when they are rubbing against something gross about the project, but I specifically tell them not to worry about fixing it, because that's how you discourage people from contributing to your codebase.

Reply:
If it were anyone else I might be more leinent and just post a bug along with this commit until I or someone else cleaned up your mess. But, any addition to file is technically a modification and this is a file you created. There's really no need to add yet another macro for toString because you won't do a simple find and replace in fear of having to change 12 lines in a file that YOU created, lol. I know you'll commit this now, forget about it and we'll run into issue later. There's reasons we started reviewing code and not just blindy taking changes from others like Robert and myself, and this is one. Honestly, If it were a new contributer you would be the one helping mitigate the issue, thus you should be the one to fix it here or as I said earlier atleast not further it.

Reply:
I think you've both made excellent points here, you're just coming from different perspectives. @fundies is just focused on the here and the now.